contracts/libraries/ArbitraryMessage.sol

```diff
@@ -0,0 +1,147 @@
+pragma solidity 0.4.24;
+
+import "../interfaces/IBridgeValidators.sol";
+
+library ArbitraryMessage {
+    function recoverAddressFromSignedMessage(bytes signature, bytes message) internal pure returns (address) {
```

Comment:
Actually only `hasEnoughValidSignatures` and `hashMessage` were different. To avoid code duplication, I added a new parameter in the methods to properly handle each case https://github.com/poanetwork/poa-bridge-contracts/pull/77/commits/d531eafc76be8fee215c0e1d80a278da868a75ec